modules/cas_disk/exp_obj.c

```diff
@@ -217,22 +217,54 @@ static int _casdsk_get_next_part_no(struct block_device *bd)
 	return part_no;
 }
 
-static int _casdsk_del_partitions(struct block_device *bd)
+static int _casdsk_del_partitions(struct casdsk_disk *dsk)
 {
-	int result = 0;
-	int part_no;
+	struct block_device *bd = casdsk_disk_get_blkdev(dsk);
+	struct file *bd_file;
+	unsigned long __user usr_bpart;
+	unsigned long __user usr_barg;
 	struct blkpg_partition bpart;
 	struct blkpg_ioctl_arg barg;
+	int result = 0;
+	int part_no;
+
+	bd_file = filp_open(dsk->path, 0, 0);
+	if (IS_ERR(bd_file))
+		return PTR_ERR(bd_file);
+
+	usr_bpart = cas_vm_mmap(NULL, 0, sizeof(bpart));
+	if (IS_ERR((void *)usr_bpart)) {
```

Comment:
cas_vm_mmap does not return error pointer. you should check for negative value instead and not use IS_ERR/PTR_ERR here

Reply:
It does. Look at the example usages in the kernel code:
https://elixir.bootlin.com/linux/latest/source/arch/x86/kvm/x86.c#L10372
https://elixir.bootlin.com/linux/latest/source/fs/binfmt_flat.c#L569